ignoreOtherParameters combined with withOutputParameter produces strange behavior

Given the following test:
```cpp
TEST(bug, ignore_output_param)
{
        int param = 1;
        int retval = 2;
        mock().expectOneCall("function")
                .withParameter("boo", 5)
                .ignoreOtherParameters();
        mock().actualCall("function")
                .withOutputParameter("parameterName", &param).withParameter("boo", 5);
        CHECK_EQUAL(param, 1);
        CHECK_EQUAL(retval, 2);
        mock().checkExpectations();
}
```

I get the following output:
```shell
 error: Failure in TEST(bug, ignore_output_param)
cpputest-3.7.2/src/CppUTestExt/MockNamedValue.cpp:190: error:
	expected <const void*>
	but was  <int>
	difference starts at position 0 at: <          int       >
	                                               ^
```

Adding `withOutputParameterReturning` causes the test to succeed, however I do not want the mock to modify the data the pointer is referring too. I just want to ignore it, which seems impossible with the built-in feature!

--- a/src/CppUTestExt/MockActualCall.cpp
+++ b/src/CppUTestExt/MockActualCall.cpp
@@ -82,9 +82,12 @@ void MockCheckedActualCall::finalizeOutputParameters(MockCheckedExpectedCall* ex
 {
     for (MockOutputParametersListNode* p = outputParameterExpectations_; p; p = p->next_)
     {
-        const void* data = expectedCall->getOutputParameter(*p->name_).getConstPointerValue();
-        size_t size = expectedCall->getOutputParameter(*p->name_).getSize();
-        PlatformSpecificMemCpy(p->ptr_, data, size);
+        MockNamedValue expectedParameter = expectedCall->getOutputParameter(*p->name_);
+        if (expectedParameter.getName() != "") {
+            const void* data = expectedCall->getOutputParameter(*p->name_).getConstPointerValue();
+            size_t size = expectedCall->getOutputParameter(*p->name_).getSize();
+            PlatformSpecificMemCpy(p->ptr_, data, size);
+        }
     }
 }
 

--- a/tests/CppUTestExt/MockSupportTest.cpp
+++ b/tests/CppUTestExt/MockSupportTest.cpp
@@ -857,6 +857,13 @@ TEST(MockSupportTest, outputParameterTraced)
     STRCMP_CONTAINS("Function name:someFunc someParameter:", mock().getTraceOutput());
 }
 
+TEST(MockSupportTest, outputParameterThatIsIgnoredShouldNotFail)
+{
+    int param;
+    mock().expectOneCall("function").ignoreOtherParameters();
+    mock().actualCall("function").withOutputParameter("parameterName", &param);
+}
+
 TEST(MockSupportTest, outputParameterWithIgnoredParameters)
 {
     int param = 1;

